plugins/plugin-logui/src/controller/kubectl/logs.ts

```diff
@@ -72,44 +43,118 @@ async function getLogsAsTable(args: Arguments<LogOptions>) {
  */
 async function doLogs(args: Arguments<LogOptions>) {
   const streamed = args.parsedOptions.follow || args.parsedOptions.f
-  const hasSelector = args.parsedOptions.selector || hasLabel(args)
-  const resourePos = args.argvNoOptions[0] === commandPrefix ? 4 : 3
-  const hasResource = args.argvNoOptions.length >= resourePos
-
-  if (!streamed && (hasResource || hasSelector)) {
-    // render as Kui table
-    return getLogsAsTable(args)
-  } else {
-    // send to PTY
-    if (streamed && !args.parsedOptions.since) {
-      // see https://github.com/kui-shell/plugin-kubeui/issues/210
-      const since = '10s'
-      args.parsedOptions.since = since
-      args.argvNoOptions.push(since)
-      args.argv.push(since)
-      args.command = args.command + ' --since=10s'
-    }
 
-    const stdout = await args.createOutputStream()
-    const myExecOptions = Object.assign({}, args.execOptions, {
-      quiet: true,
-      replSilence: true,
-      echo: false,
-      onInit: (ptyJob: Abortable) => {
-        return _ => {
-          if (args.block['isCancelled']) {
-            ptyJob.abort()
+  // if we are streaming (logs -f), and the user did not specify a
+  // "--since", then add one, to prevent the default behavior of
+  // kubectl which fetches quite a bit of history
+  if (streamed && !args.parsedOptions.since) {
+    // see https://github.com/kui-shell/plugin-kubeui/issues/210
+    const since = '10s'
+    args.parsedOptions.since = since
+    args.argv.push('--since=since')
+    args.command = args.command + ' --since=10s'
+  }
+
+  // if we are not streaming, and the user has not specified a
+  // "--tail", then add one, for the same reason
+  if (!streamed && !args.parsedOptions.tail) {
+    const tail = 30
+    args.parsedOptions.tail = tail
+    args.argv.push('--tail=' + tail)
+    args.command = args.command + ' --tail=' + tail
+  }
+
+  // set up the PTY stream; we want to stream to this stdout sink
+  const stdout = await args.createOutputStream()
+
+  // a bit of plumbing: tell the PTY that we will be handling everything
+  const myExecOptions = Object.assign({}, args.execOptions, {
+    rethrowErrors: true, // we want to handle errors
+    quiet: true, // don't ever emit anything on your own
+    replSilence: true, // repl: same thing
+    echo: false, // do not even echo "ok"
+
+    // the PTY will call this when the PTY process is ready; in
+    // return, we send it back a consumer of streaming output
+    onInit: (ptyJob: Abortable) => {
+      let curLine: string
+
+      // _ is one chunk of streaming output
+      return async (_: Streamable) => {
+        if (args.block['isCancelled']) {
+          ptyJob.abort()
+        } else if (typeof _ === 'string') {
+          // we only know how to handle strings
+          if (/\n$/.test(_)) {
+            const joined = curLine ? curLine + _ : _
+            // eslint-disable-next-line no-control-regex
+            const fullLine = /\x1b/.test(joined)
```

Comment:
why it's checking `\x1b` here? 
so if there is `\x1b`, don't do any color replace, but use ansi directly?

Reply:
i meant to add a comment, thanks for the poke. the check here is: if the output from the PTY already contains ANSI control characters, then we won't add any of our own. \x1b is ESCAPE